partiql-cli/src/main/kotlin/org/partiql/cli/pipeline/ErrorMessageFormatter.kt

```diff
@@ -74,6 +80,69 @@ object ErrorMessageFormatter {
         return "Variable reference$idStr could not be found in the database environment or in the set of available locals.$localsStr"
     }
 
+    /**
+     * @see PError.INVALID_EXCLUDE_PATH
+     */
+    private fun invalidExcludePath(error: PError): String {
+        val path = error.getOrNull("PATH", String::class.java)
+        val pathStr = prepare(path, " (", ")")
+        return "Invalid exclude path$pathStr."
+    }
+
+    /**
+     * @see PError.CARDINALITY_VIOLATION
+     */
+    private fun cardinalityViolation(): String {
+        return "Cardinality violation."
+    }
+
+    /**
+     * @see PError.NUMERIC_VALUE_OUT_OF_RANGE
+     */
+    private fun numericValueOutOfRange(error: PError): String {
+        val value = error.getOrNull("VALUE", String::class.java)
+        val valueStr = prepare(value.toString(), " (", ")")
+        val type = error.getOrNull("TYPE", PType::class.java)
+        val typeString = prepare(type.toString(), "for type ")
+        return "Numeric value$valueStr is out of range$typeString."
```

Comment:
spacing is slightly off

```suggestion
        val typeString = prepare(type.toString(), " for type ")
        return "Numeric value$valueStr is out of range$typeString."
```

Reply:
may be better (for readability and consistency) to have the returned string template control the whitespace rather than the inner values:

e.g. 

```suggestion
        val typeString = prepare(type.toString(), "for type ")
        return "Numeric value $valueStr is out of range $typeString."
```
where `$valueStr` and `$typeString` don't have the space prefix.

Reply:
Agree with the first comment.

The second comment wouldn't work if the type/value were null (which they can be). The string would look like:
```
Numeric value  is out of range .
```